pkg/reconcilers/serviceaccount.go

```diff
@@ -0,0 +1,70 @@
+package reconcilers
+
+import (
+	"fmt"
+	"reflect"
+	"sort"
+	"strings"
+
+	"github.com/3scale/3scale-operator/pkg/common"
+	v1 "k8s.io/api/core/v1"
+)
+
+func ServiceAccountImagePullPolicyMutator(existingObj, desiredObj common.KubernetesObject) (bool, error) {
+	existing, ok := existingObj.(*v1.ServiceAccount)
+	if !ok {
+		return false, fmt.Errorf("%T is not a *v1.ServiceAccount", existingObj)
+	}
+	desired, ok := desiredObj.(*v1.ServiceAccount)
+	if !ok {
+		return false, fmt.Errorf("%T is not a *v1.ServiceAccount", desiredObj)
+	}
+
+	updated := false
+
+	// Copy received ServiceAccounts due to we are going to
+	// sort (modify) its ImagePullSecrets
```

Comment:
Sorting allows us to do the comparison. However, it means that `.spec.ImagePullSecrets` array positions do not matter when reconciliating.

```
- a
- b
```

vs

```
- b
- a

Will not be detected as a change and will not trigger any reconciliation

Reply:
no issue, order is not important. The CRD type could be a map also to add some "secret" name at the CR level.

Reply:
I'll leave it as array to match how is defined in K8s